volttron/utils/rmq_setup.py

```diff
@@ -488,120 +508,93 @@ def _verify_and_save_instance_ca(instance_ca_path, instance_ca_key):
         # generate a new rabbitmq_config.yml
-        _create_rabbitmq_config(instance_name, type)
+        _create_rabbitmq_config(rmq_config, setup_type)
 
     # Load either the newly created config or config passed
-    error = _load_rmq_config()
-    if error:
+    try:
+        rmq_config.load_rmq_config()
+    except (IOError, yaml.YAMLError) as exc:
         _log.error("\nFor single setup, configuration file must at least "
                    "contain host and ssl certificate details. For federation "
                    "and shovel setup, config should contain details about the "
                    "volttron instance with which communication needs "
                    "to be established. Please refer to example config file "
                    "at examples/configurations/rabbitmq/rabbitmq_config.yml")
-        return error
+        return exc
 
     invalid = True
-    if type in ["all", "single"]:
+    if setup_type in ["all", "single"]:
         invalid = False
-        _start_rabbitmq_without_ssl()
+        _start_rabbitmq_without_ssl(rmq_config)
         _log.debug("Creating rabbitmq virtual hosts and required users for "
                    "volttron")
         # Create local RabbitMQ setup - vhost, exchange etc.
         # should be called after _start_rabbitmq_without_ssl
         rmq_mgmt = RabbitMQMgmt()
         success = rmq_mgmt.init_rabbitmq_setup()
-        ssl_auth = config_opts.get('ssl', 'true')
-        if success and ssl_auth in ('true', 'True', 'TRUE', True):
-            _setup_for_ssl_auth(instance_name)
-
-        # Create utility scripts
-        script_path = os.path.dirname(os.path.realpath(__file__))
-        src_home = os.path.dirname(os.path.dirname(script_path))
-        start_script = os.path.join(src_home, 'start-rabbitmq')
-        with open(start_script, 'w+') as f:
-            f.write(os.path.join(config_opts['rmq-home'],'sbin',
-                                 'rabbitmq-server') + ' -detached')
-            f.write(os.linesep)
-            f.write("sleep 5")  # give a few seconds for all plugins to be ready
-        os.chmod(start_script, 0o755)
-
-        stop_script = os.path.join(src_home, 'stop-rabbitmq')
-        with open(stop_script, 'w+') as f:
-            f.write(os.path.join(config_opts['rmq-home'], 'sbin',
-                                 'rabbitmqctl') + ' stop')
-        os.chmod(stop_script, 0o755)
-
-        #symlink to rmq log
-        log_name = os.path.join(src_home, 'rabbitmq.log')
-        if os.path.lexists(log_name):
-            os.unlink(log_name)
-        os.symlink(os.path.join(config_opts['rmq-home'],
-                                'var/log/rabbitmq',
-                                "rabbit@" +
-                                config_opts['host'].split('.')[0] + ".log"),
-                   log_name)
-
-
-
-    if type in ["all", "federation"]:
+        if success and rmq_config.is_ssl:
+            _setup_for_ssl_auth(rmq_config)
+
```

Comment:
Why is logic to create utility scripts deleted? Is this moved somewhere else?

Reply:
Oops! Must be a merge problem.